docker/run-uc/Dockerfile

```diff
@@ -0,0 +1,109 @@
+FROM eclipse-temurin:22-jre-alpine AS run_uc
```

Comment:
I took the liberty of creating a multi-stage `Dockerfile` (for the root directory):

```
FROM eclipse-temurin:21-jdk AS build_uc

COPY . /unitycatalog/
WORKDIR /unitycatalog/

RUN ./build/sbt assembly

FROM eclipse-temurin:21-jre-alpine AS run_uc

ARG unitycatalog_uid=185
ARG unitycatalog_home="/opt/unitycatalog"
ARG unitycatalog_jar="server/target"
ARG unitycatalog_etc="etc"
ARG unitycatalog_user_name="unitycatalog"
ARG unitycatalog_user_home="home"
ARG unitycatalog_user_basedir="${unitycatalog_home}/${unitycatalog_user_home}"

ENV SERVER_TARGET_DIR="${unitycatalog_home}/${unitycatalog_jar}"

# Create the UC directories
RUN <<-EOF
    set -ex;
    mkdir -p "${unitycatalog_home}";
    mkdir -p "${unitycatalog_home}/${unitycatalog_jar}";
    mkdir -p "${unitycatalog_home}/${unitycatalog_etc}";
    mkdir -p "${unitycatalog_home}/${unitycatalog_user_home}";
EOF


# Create system group and user for Unity Catalog
# Ensure the user created has their HOME pointing to the volume
# created to persist user data and the sbt cached files that
# are created as a result of compiling the unity catalog.
# This also ensures that the container can run independently from
# the storage, so we can have ephemeral docker instances with --rm
# and still be able to run the unity catalog each time without problems.
RUN <<-EOF
    #!/usr/bin/env bash
    set -ex;
    addgroup --system --gid "${unitycatalog_uid}" "${unitycatalog_user_name}";
    adduser --system --uid "${unitycatalog_uid}" \
            --ingroup "${unitycatalog_user_name}" \
            --home "${unitycatalog_user_basedir}" \
            --shell "$(/usr/bin/env bash)" \
            "${unitycatalog_user_name}";
EOF

# Define volume to persist Unity Catalog data
VOLUME  ${unitycatalog_home}

WORKDIR "$unitycatalog_home"

# Copy the Uber-Jar from the previous stage
# Full path: /unitycatalog/server/target/unitycatalog-server-assembly-0.1.0-SNAPSHOT.jar
COPY --from=build_uc "/unitycatalog/${unitycatalog_jar}/*.jar" "${unitycatalog_home}/${unitycatalog_jar}/"

# Copy the etc folder which contains the config files and the data folder
COPY ./etc/ "${unitycatalog_home}/${unitycatalog_etc}/"

# Set ownership of directories and Unity Catalog home directory to a less
# priviledged user
RUN <<-"EOF"
    #!/usr/bin/env bash
    set -ex;
    chown -R "${unitycatalog_user_name}":"${unitycatalog_user_name}" "$unitycatalog_home";
    chmod +x /opt/unitycatalog/server/target/*
EOF

# Set the user for the container process
USER "${unitycatalog_user_name}"

CMD ["java", "-jar", "/opt/unitycatalog/server/target/unitycatalog-server-assembly-0.1.0-SNAPSHOT.jar"]
```

Let me know what your thoughts are!